front/src/common/RollingStockSelector/RollingStockCardDetail.tsx

```diff
@@ -40,34 +56,42 @@ export default function RollingStockCardDetail(props) {
       };
       // AC & HEATING curves (optional)
       curvesData.modes[mode].curves.forEach((curve) => {
-        const serieId = `${name} ${curve.cond.comfort}`;
-        transformedCurves[serieId] = { ...curve.curve, mode: name, comfort: curve.cond.comfort };
+        if (curve.cond) {
+          const serieId = `${name} ${curve.cond.comfort}`;
+          transformedCurves[serieId] = {
+            ...curve.curve,
+            mode: name,
+            comfort: curve.cond.comfort as string,
+          };
+        }
       });
     });
     return transformedCurves;
   };
 
-  const getRollingStock = async () => {
-    try {
-      const rollingStockData = await get(`${ROLLING_STOCK_URL + id}/`);
-      setData(rollingStockData);
-      setCurvesComfortList(listCurvesComfort(rollingStockData.effort_curves));
-    } catch (e) {
+  const { data, error } = osrdEditoastApi.useGetRollingStockByIdQuery(
```

Comment:
```suggestion
  const { data: rollingStock, error } = osrdEditoastApi.useGetRollingStockByIdQuery(
```
You can rename the result of the api this way ;)

Reply:
done